druntime/src/core/internal/vararg/aarch64.d

```diff
@@ -36,7 +36,11 @@ extern (C++, std) struct __va_list
 }
 
 ///
-alias va_list = __va_list;
+alias __va_list = __va_list_tag;
+
+///
+alias va_list = __va_list*;
```

Comment:
This is wrong too (mangling again).

Reply:
This is *not* resolved - you're breaking mangling with `extern(C++)`, for e.g. https://github.com/dlang/dmd/blob/f0541d65ba777e6f03499bcb5c0c59da8ce94050/compiler/test/runnable_cxx/cppa.d#L327-L331

As said in the other PR, GDC and LDC already work for AArch64, there's nothing to fix in druntime for them. So everything you change or add and isn't guarded by `version(DigitalMars)` is very likely to cause a regression for the other compilers.

That also goes for the `__va_list_tag` symbol - why does DMD need it?

Reply:
And yeah, the AAPCS64 `va_list` is very special, see my comment in https://github.com/ldc-developers/ldc/blob/3acd7617fcfee90470adfeb99da8b9f06a67ca01/gen/abi/aarch64.cpp#L22-L29. It is mangled as a `std::__va_list` struct *value*, but only ever passed by reference under the hood.

Reply:
__va_list_tag appears in the AArch64 ABI specification pdf, which is why I added it.

I didn't know about that special case for va_list. Just shoot me now.

Reply:
__va_list_tag is used in stdc/stdarg.d

Reply:
I do plan on getting a Mac mini AArch64. But not just yet.

Reply:
BTW the AArch64 supports 128 bit floats.